drivers/io/io_fip.c

```diff
@@ -28,16 +32,22 @@
 		x.node[4], x.node[5]
 
 typedef struct {
+	void *dev_spec;
 	unsigned int file_pos;
 	fip_toc_entry_t entry;
 } file_state_t;
 
 static const uuid_t uuid_null = {0};
-static file_state_t current_file = {0};
+
+static file_state_t current_file_pool[MAX_FIP_DEVICES];
+static io_dev_info_t dev_info_pool[MAX_FIP_DEVICES];
+
+/* Track number of allocated fip devices */
+static unsigned int fip_dev_count;
+
 static uintptr_t backend_dev_handle;
```

Comment:
There is only one instance of backend_dev_handle. Shouldn't there be multiple instances of this as well ?

Reply:
I was working on the case where all the fip devices had the same backend handle. But , from driver perspective we should not limit this. I will move these handles into the info structure.